build_defs/java_opts.bzl

```diff
@@ -1,22 +1,43 @@
 """Java options and protobuf-specific java build rules with those options."""
 
-load("@rules_java//java:defs.bzl", "java_library")
+load("@rules_java//java:defs.bzl", "java_library", "java_import")
 load("@rules_jvm_external//:defs.bzl", "java_export")
 load("//java/osgi:osgi.bzl", "osgi_java_library")
 load("//:protobuf_version.bzl", "PROTOBUF_JAVA_VERSION")
 
+BUNDLE_DOC_URL = "https://developers.google.com/protocol-buffers/"
+BUNDLE_LICENSE = "https://opensource.org/licenses/BSD-3-Clause"
+
+## JDK target to use for versioned classes.
+JPMS_JDK_TARGET = "9"
+
+## Java compiler options for JDK 8.
 JAVA_OPTS = [
     "-source 8",
     "-target 8",
     "-Xep:Java8ApiChecker:ERROR",
 ]
 
-BUNDLE_DOC_URL = "https://developers.google.com/protocol-buffers/"
-BUNDLE_LICENSE = "https://opensource.org/licenses/BSD-3-Clause"
+## Java compiler options for JDK 9+.
+JAVA9_OPTS = [
+    "-source 9",
+    "-target 9",
+]
+
+## Default JAR manifest entries to stamp on export.
+DEFAULT_MANIFEST_ENTRIES = {
+    "Multi-Release": "true"
+}
+
+## Default visibility settings for Java module targets.
+DEFAULT_JMOD_VISIBILITY = [
+    "//java:__subpackages__",
+]
```

Comment:
1) Java 9 module opts
2) JAR manifests should carry `Multi-Release: true` when using `META-INF/versions/...`
3) Java Module targets built in this PR are private to `//java/...`